multicluster/controllers/multicluster/stale_controller.go

```diff
@@ -270,6 +297,38 @@ func (c *StaleResCleanupController) cleanupClusterInfoResourceExport(commonArea
 	return nil
 }
 
+func (c *StaleResCleanupController) cleanupMemberClusterAnnounces() error {
+	memberClusterAnnounceList := &mcsv1alpha1.MemberClusterAnnounceList{}
+	if err := c.List(ctx, memberClusterAnnounceList, &client.ListOptions{}); err != nil {
+		return err
+	}
+
+	for _, m := range memberClusterAnnounceList.Items {
+		var needToDelete bool
+		memberClusterAnnounce := m
+
+		lastUpdateTime, err := time.Parse(time.RFC3339, memberClusterAnnounce.Annotations[commonarea.TimestampAnnotationKey])
+		if err == nil && time.Now().Sub(lastUpdateTime) < memberClusterAnnounceStaleTime {
+			needToDelete = false
```

Comment:
You can do "continue" here, and then you need not "needToDelete". Up to you.

Reply:
Removed the "needToDelete" and use continue.